river/metrics/rand.py

```diff
@@ -0,0 +1,175 @@
+from river import metrics
+
+from . import base
+
+__all__ = ["AdjustedRand", "Rand"]
+
+
+class Rand(base.MultiClassMetric):
+    """Rand Index.
+
+    The Rand Index [^1] [^2] is a measure of the similarity between two data clusterings.
+    Given a set of elements `S` and two partitions of `S` to compare, `X` and `Y`,
+    define the following:
+
+    * a, the number of pairs of elements in `S` that are in the **same** subset in `X`
+    and in the **same** subset in `Y`
+
+    * b, the number of pairs of elements in `S` that are in the **different** subset in `X`
+    and in **different** subsets in `Y`
+
+    * c, the number of pairs of elements in `S` that are in the **same** subset in `X`
+    and in **different** subsets in `Y`
+
+    * d, the number of pairs of elements in `S` that are in the **different** subset in `X`
+    and in the **same** subset in `Y`
+
+    The Rand index, R, is
+
+    $$
+    R = \frac{a+b}{a+b+c+d} = \frac{a+b}{\frac{n(n-1)}{2}}.
+    $$
+
+    Parameters
+    ----------
+    cm
+        This parameter allows sharing the same confusion
+        matrix between multiple metrics. Sharing a confusion matrix reduces the amount of storage
+        and computation time.
+
+    Examples
+    --------
+    >>> from river import metrics
+
+    >>> y_true = [1, 1, 2, 2, 3, 3]
+    >>> y_pred = [1, 1, 1, 2, 2, 2]
+
+    >>> metric = metrics.Rand()
+
+    >>> for yt, yp in zip(y_true, y_pred):
+    ...     print(metric.update(yt, yp).get())
+    1.0
+    1.0
+    0.3333333333333333
+    0.5
+    0.6
+    0.6666666666666666
+
+    >>> metric
+    Rand: 0.666667
+
+    References
+    ----------
+    [^1]: Wikipedia contributors. (2021, January 13). Rand index.
+          In Wikipedia, The Free Encyclopedia,
+          from https://en.wikipedia.org/w/index.php?title=Rand_index&oldid=1000098911
+    [^2]: W. M. Rand (1971). "Objective criteria for the evaluation of clustering methods".
+          Journal of the American Statistical Association. American Statistical Association.
+          66 (336): 846–850. arXiv:1704.01036. doi:10.2307/2284239. JSTOR 2284239.
+
+    """
+
+    def __init__(self, cm=None):
+        super().__init__(cm)
+
+    @property
+    def works_with_weights(self):
+        return False
+
+    def get(self):
+
+        pair_confusion_matrix = metrics.PairConfusionMatrix(self.cm).get()
+
+        true_positives = pair_confusion_matrix[1][1]
+        true_negatives = pair_confusion_matrix[0][0]
+
+        total_pairs = self.cm.n_samples * (self.cm.n_samples - 1)
+
+        try:
+            return (true_positives + true_negatives) / total_pairs
+        except ZeroDivisionError:
+            return 1.0
```

Comment:
Why 1.0? Should this metric define:

```python
@property
    def bigger_is_better(self):
        return False
```

Reply:
When the total number of pairs is zero, the rand index is automatically set to `1.0`. This is done so that the results are 100% in line with the implementation of the rand index in scikit-learn. 
And yes, the `bigger_is_better` should definitely be implemented!

Reply:
Understood, thanks for the clarification.